rules/template-indent.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+const stripIndent = require('strip-indent');
+const indentString = require('indent-string');
+const esquery = require('esquery');
+const {replaceTemplateElement} = require('./fix/index.js');
+
+const MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE = 'template-indent';
+const messages = {
+	[MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE]: 'Templates should be properly indented.',
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const sourceCode = context.getSourceCode();
+	const options = {
+		tags: ['outdent', 'dedent', 'gql', 'sql', 'html', 'styled'],
+		functions: ['dedent', 'stripIndent'],
+		selectors: [],
+		comments: ['HTML', 'indent'],
+		...context.options[0],
+	};
+
+	options.comments = options.comments.map(comment => comment.toLowerCase());
+
+	const selectors = [
+		...options.tags.map(tag => `TaggedTemplateExpression[tag.name="${tag}"] > .quasi`),
+		...options.functions.map(fn => `CallExpression[callee.name="${fn}"] > .arguments:first-child`),
```

Comment:
We should mention that only the first child is checked, or make change here to check all arguments.

Personally I prefer check all arguments.

Reply:
Done, and added a test case.